packages/catalog-model/src/entity/policies/DefaultParentEntityPolicy.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '../Entity';
+import { GroupEntity } from '../../kinds';
+import { EntityPolicy } from './types';
+import { DEFAULT_NAMESPACE } from '../constants';
+import { parseEntityRef, stringifyEntityRef } from '../ref';
+
+/**
+ * DefaultParentPolicy is an EntityPolicy that updates group entities
+ * with a parent of last resort. This ensures that, while we preserve
+ * any existing group hierarchies, we can guarantee that there is a
+ * single global root of the group hierarchy.
+ *
+ * @public
+ */
+export class DefaultParentEntityPolicy implements EntityPolicy {
+  private readonly parentRef: string;
+
+  constructor(parent: string) {
+    const { kind, namespace, name } = parseEntityRef(parent, {
+      defaultKind: 'group',
+      defaultNamespace: DEFAULT_NAMESPACE,
+    });
+
```

Comment:
Should we actually add a check here that `kind.toLocaleLowerCase('en-US') === 'group'`? Just to protect against really silly inputs :)

Reply:
I always thought that the kind was case-insensitive, though I never bothered to dig around and find where that happens.

Reply:
`stringifyEntityRef` does this, so I don't think we need it here.

Reply:
Well the reason I mentioned it was because of the defaultKind also being meant to be uppercased.

I think one might say - there's no case _insensitivity_ per se anywhere. Refs when serialized on string form are per definition lowercased, that's all. In other places the casing should in general be left alone. If somebody tries to ingest an entity with the wrong casing in the kind, it will be rejected.

Reply:
OK, I added the same check used by `isGroupEntity`.